### DISCUSSION TITLE
How to check if the Entry field type relation is empty in antlers?

### DISCUSSION THREAD

**Discussion by JeremyWorkWork** [statamic/cms#8705] [Troubleshooting] [answered]
I have a collection called Products
I have a collection called Categories

Products has a Entry field type relation

```-
  handle: categories
  field:
    type: entries
    taxonomies:
      - categories
    display: categories
    mode: default
    create: true
    collections:
      - categories
    icon: entries
    listable: hidden
    instructions_position: above
    visibility: visible
    hide_display: false
```

In my antlers template for the product collection:
```
{{ categories }}
    {{ title }}
{{ /categories }}
```

If the Product collection **has** related categories then the title is of the **category title**, if there are no related categories then the title is of the **product**.

Categories always evaluates to true even if there are no relations. Hence the above outputs the product title if there are no related categories

How can i loop the related categories alway as an array or at least check if there are no related categories?

Do i need to use something like this to check the raw value first?
```{{ if categories | raw }}```

is_empty modifier doesn't seem to help.

dumping out $categories when there are no relations shows the Statamic\Fields\Value instance with a raw value of null.

dumping out $categories when there are relations shows the Statamic\Fields\Value instance with a raw value with the array of related entry ids.

**Comment by ryanmitchell** [comment, marked as answer]
`{{ if (categories | count) > 0 }}` should work for this

**Reply by JeremyWorkWork** [reply]
Thanks, works. I missed that.

Also possible to simplify to {{ if products | count }}
